benches/benches/set/blockchain.rs

```diff
@@ -32,6 +41,73 @@ use rand::{
     SeedableRng,
 };
 
+/// Reimplementation of `tempdir::TempDir` that allows creating a new
+/// instance without actually creating a new directory on the filesystem.
+/// This is needed since rocksdb requires empty directory for checkpoints.
+pub struct ShallowTempDir {
+    path: PathBuf,
+}
+impl ShallowTempDir {
+    pub fn new() -> Self {
+        let mut rng = rand::thread_rng();
+        let mut path = env::temp_dir();
+        path.push(format!("fuel-core-bench-rocksdb-{}", rng.next_u64()));
+        Self { path }
+    }
+}
+impl Drop for ShallowTempDir {
+    fn drop(&mut self) {
+        // Ignore errors
+        let _ = std::fs::remove_dir_all(&self.path);
+    }
+}
+
+pub struct BenchDb {
+    db: RocksDb,
+    /// Used for RAII cleanup. Contents of this directory are deleted on drop.
+    tmp_dir: ShallowTempDir,
+}
+
+impl BenchDb {
+    fn new(contract: &ContractId) -> io::Result<Self> {
+        let tmp_dir = ShallowTempDir::new();
+        let db = RocksDb::default_open(&tmp_dir.path, None).unwrap();
+        let db = Arc::new(db);
+        let mut database = Database::new(db.clone());
+        database.init_contract_state(
```

Comment:
For some, opcodes we need to init balances too(for example `SMO` or `MINT` or `BURN`)